tests/generated_data/generate_md.py

```diff
@@ -0,0 +1,153 @@
+"""Script for generating new metadata files."""
+
+# Copyright New York University and the TUF contributors
+# SPDX-License-Identifier: MIT OR Apache-2.0
+
+import os
+import sys
+from datetime import datetime
+from typing import Any, Dict, Optional
+
+from securesystemslib.signer import SSlibSigner
+
+from tests import utils
+from tuf.api.metadata import (
+    SPECIFICATION_VERSION,
+    TOP_LEVEL_ROLE_NAMES,
+    Key,
+    Metadata,
+    MetaFile,
+    Role,
+    Root,
+    Snapshot,
+    Targets,
+    Timestamp,
+)
+from tuf.api.serialization.json import JSONSerializer
+
+# Hardcode keys and expiry time to achieve reproducibility.
+keys_dict: Dict[str, Dict[str, Any]] = {
+    "ed25519_1": {
+        "keytype": "ed25519",
+        "scheme": "ed25519",
+        "keyid": "5822582e7072996c1eef1cec24b61115d364987faa486659fe3d3dce8dae2aba",
+        "keyval": {
+            "public": "b11d2ff132c033a657318c74c39526476c56de7556c776f11070842dbc4ac14c",
+            "private": "510e5e04d7a364af850533856eacdf65d30cc0f8803ecd5fdc0acc56ca2aa91c",
+        },
+    },
+    "ed25519_2": {
+        "keytype": "ed25519",
+        "scheme": "ed25519",
+        "keyid": "09d440e3725cec247dcb8703b646a87dd2a4d75343e8095c036c32795eefe3b9",
+        "keyval": {
+            "public": "250f9ae3d1d3d5c419a73cfb4a470c01de1d5d3d61a3825416b5f5d6b88f4a30",
+            "private": "e6645b00312c8a257782e3e61e85bafda4317ad072c52251ef933d480c387abd",
+        },
+    },
+    "ed25519_3": {
+        "keytype": "ed25519",
+        "scheme": "ed25519",
+        "keyid": "3458204ed467519c19a5316eb278b5608472a1bbf15850ebfb462d5315e4f86d",
+        "keyval": {
+            "public": "82380623abb9666d4bf274b1a02577469445a972e5650d270101faa5107b19c8",
+            "private": "cd13dd2180334b24c19b32aaf27f7e375a614d7ba0777220d5c2290bb2f9b868",
+        },
+    },
+    "ed25519_4": {
+        "keytype": "ed25519",
+        "scheme": "ed25519",
+        "keyid": "2be5c21e3614f9f178fb49c4a34d0c18ffac30abd14ced917c60a52c8d8094b7",
+        "keyval": {
+            "public": "0e6738fc1ac6fb4de680b4be99ecbcd99b030f3963f291277eef67bb9bd123e9",
+            "private": "7e2e751145d1b22f6e40d4ba2aa47158207acfd3c003f1cbd5a08141dfc22a15",
+        },
+    },
+}
```

Comment:
You could condense this a bit, but having a list of (keyid, public val, private val)-tuples and construct the dicts below in a loop. You could even omit keyid in the tuple and compute it in the loop.

Reply:
I followed your suggestions and made three lists `public_values`, `private_values` and `keyids`.
I don't want to calculate the `keyids` as that is a possible place where a change inside `securesystemslib` will result in our tests failing which I am not sure we want.